### ISSUE TITLE
Resource Elasticsearch fails with undefined method `[]' for nil:NilClass

### ISSUE THREAD

**Issue by valentinvsbg** [inspec/inspec#3916]
### 🎛 Description
Elasticsearch resource throws exception 
```
undefined method `[]' for nil:NilClass
```
in the following cases:
1) when the private resource function `verify_json_payload` is called.
2) when there are no elasticsearch modules or plugins on a given node.

The first issue happening because the private function checks for a property that does not exist. At https://github.com/inspec/inspec/blob/master/lib/resources/elasticsearch.rb#L164
you may notice `if content['_nodes']['successful'].zero?` There is no such property `_nodes` or `successful`  in the object.

The second issue is because the elasticsearch has no modules in use thus the node data has no such collection. At https://github.com/inspec/inspec/blob/master/lib/resources/elasticsearch.rb#L107 
attempts to map the modules but in my case there is no such property `modules`. The plugins however is populated as expected, its just that we don't use elasticsearch modules I guess.

🐛 bug report

### 🌍 InSpec and Platform Version
inspec - 3.9.0v
ruby - 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]
Ubuntu 18.04.1 LTS
Elasticsearch:
"version" : {
    "number" : "1.7.1",
    "build_hash" : "b88f43fc40b0bcd7f173a1f9ee2e97816de80b19",
    "build_timestamp" : "2015-07-29T09:54:16Z",
    "build_snapshot" : false,
    "lucene_version" : "4.10.4"
  }

### 🤔 Replication Case
When you run a simple test against elasticsearch which has no modules in use.
```
  describe elasticsearch do
    its('node_name') { should include 'xxxxxx' }
  end
```
then the test will get skipped with the error:
`undefined method `[]' for nil:NilClass`

### 💁 Possible Solutions
Implement simple check for elasticsearch modules and plugins at:
https://github.com/inspec/inspec/blob/master/lib/resources/elasticsearch.rb#L107

I have copied the resource locally and implemented a check right before the mapping. This way the resource was working fine:
```
       if(node.modules != nil)
          node.module_list = node.modules.map(&:name)
        else
          node.module_list = []
        end
```
Also I had to remove the line at https://github.com/inspec/inspec/blob/master/lib/resources/elasticsearch.rb#L164 

I believe that elasticsearch plugins and modules are an optional thing so there must be a check to ensure no exceptions are thrown.

**Comment by Vasu1105** [comment]
I am unable to reproduce this issue. When I configured elasticsearch it by default added modules into it and also the plugins array is coming as empty. (basically not able to find the case where plugins and modules will be nil).  I tried to find out how I can configure elasticsearch without modules but did not find any specific doc for it. So giving more details on how you configured elasticsearch without modules will help to reproduce this issue or add this check blindly pr we can close this issue as it's not reproducible.
